src/Disks/DiskLocal.cpp

```diff
@@ -292,6 +296,28 @@ void DiskLocal::copy(const String & from_path, const std::shared_ptr<IDisk> & to
         IDisk::copy(from_path, to_disk, to_path); /// Copy files through buffers.
 }
 
+int DiskLocal::open(const String & path, mode_t mode) const
+{
+    String full_path = disk_path + path;
+    int fd = ::open(full_path.c_str(), mode);
+    if (-1 == fd)
+        throwFromErrnoWithPath("Cannot open file " + full_path, full_path,
+                        errno == ENOENT ? ErrorCodes::FILE_DOESNT_EXIST : ErrorCodes::CANNOT_OPEN_FILE);
+    return fd;
+}
+
+void DiskLocal::close(int fd) const
+{
+    if (-1 == ::close(fd))
+        throw Exception("Cannot close file", ErrorCodes::CANNOT_CLOSE_FILE);
```

Comment:
errors of such types MUST be handled as `abort()` becasue it means hard programming error and possibly FD leakage,